tests/std/tests/P1614R2_spaceship/test.cpp

```diff
@@ -231,6 +232,26 @@ void diagnostics_test() {
     }
 }
 
+template <bool Equal>
+struct compare_resource final : std::pmr::memory_resource {
+private:
+    void* do_allocate(size_t, size_t) override {
+        throw std::bad_alloc{};
+    }
+    void do_deallocate(void*, size_t, size_t) override {}
+    bool do_is_equal(const memory_resource&) const noexcept override {
+        return Equal;
+    }
+};
+
+template <bool Equal>
+struct compare_allocator : public std::allocator<int> {};

```

Comment:
This doesn't meet the Cpp17Allocator requirements because of rebinding. `std::allocator` provides a legacy `rebind` struct, which will report the wrong type when inherited, and `compare_allocator` lacks a rebinding constructor.

Reply:
I think this actually would have been fine since this test is C++20-only and doesn't define the `_HAS_DEPRECATED_ALLOCATOR_MEMBERS` escape hatch. :shrug: